frontend/main.js

```diff
@@ -50,59 +51,40 @@ async function startApp () {
   // force debug output even in production
   const debugParam = new URLSearchParams(window.location.search).get('debug')
   if (process.env.NODE_ENV !== 'production' || debugParam === 'true') {
-    const reducer = (o, v) => { o[v] = true; return o }
```

Comment:
This should remain unchanged. The problem is that Set is `O(n)` while maps are `O(1)`, and we need these to be as fast as possible since they will get triggered on just about every SBP invocation.

Reply:
Most sources I could find say that `Set.has()` is `O(1)` in time. I've used sets here instead of arrays precisely to avoid the `O(n)` cost of array `includes()`.

Reply:
AFAICT, [Set.has is `O(n)`](https://stackoverflow.com/a/31092145/1781435).

>  to avoid the O(n) cost of Array includes().

... We aren't using includes()...?

Reply:
> ~~AFAICT, [Set.has is `O(n)`](https://stackoverflow.com/a/31092145/1781435).~~

Oops! I must be tired. I misread that. Here's another interesting comment that includes benchmarks: https://stackoverflow.com/a/69125167/1781435

I got slightly different results for Firefox:

<img width="340" alt="Screen Shot 2022-07-15 at 8 42 26 PM" src="https://user-images.githubusercontent.com/138706/179216582-d78dcf25-d7a7-47d1-9876-883dd23a0058.png">

It shows that `Object#hasOwnProperty` is faster than `Set.has`, sometimes by an order of magnitude. So we should stick with that. It's not surprising that should be so, because a Set is considered to be an array with checks, whereas an object/map is considered to be a hashtable with O(1) lookup.

Reply:
If I understand correctly (which is a bit tricky here), according to this SO answer the spec mandates Set `has()` to be sublinear, i.e. `O(n)` _at worst_. So implementors are free to make it better, which should be feasible since Set implementations likely use a hashtable internally.

However, now that I think about it... remembering that JS sets are _ordered_ sets and not just plain sets, makes me have doubts about the efficiency of `has()`. Perhaps this constraint prevents it to be as fast as object property access (I would be kinda disappointed though).

So in doubt I will agree to revert this change.

>... We aren't using includes()...? 

True. I would have had to use it after removing the reducers, if I had let the blacklists be plain arrays. So I opted for sets.

Reply:
Just read your latest comment. So yeah let's stick with plain objects.